### ISSUE TITLE
Regression with `MappedColumn` in SQLAlchemy

### ISSUE THREAD

**Issue by adampauls** [microsoft/pyright#5107]
The code from [this documentation](https://docs.sqlalchemy.org/en/20/orm/dataclasses.html) now issues an error as of 1.308. 

**Describe the bug**

**To Reproduce**
```python
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column
from sqlalchemy.orm import MappedAsDataclass


class Base(MappedAsDataclass, DeclarativeBase):
    """subclasses will be converted to dataclasses"""


class User(Base):
    __tablename__ = "user_account"

    id: Mapped[int] = mapped_column(init=False, primary_key=True)
    name: Mapped[str]
```

**Expected behavior**
Should typecheck.

**Screenshots or Code**
```
error: Expression of type "MappedColumn[_T@Mapped]" cannot be assigned to declared type "Mapped[int]"
 "MappedColumn[_T@Mapped]" is incompatible with "Mapped[int]"
      TypeVar "_T@Mapped" is invariant
        Type "_T@Mapped" cannot be assigned to type "int" (reportGeneralTypeIssues)
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Command line.

**Additional context**
Works fine with 1.307.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a very old bug that was unmasked by a recent change. It will be addressed in the next release.

**Comment by adampauls** [comment]
If you'll permit the meme: death, taxes, and @erictraut fixing a bug within 24 hours of it being reported. Thanks!

**Comment by erictraut** [comment]
This is included in pyright 1.1.309, which I just published. It will also be included in a future release of pylance.
